toolkit/tools/pkg/imagecustomizerlib/customizeutils.go

```diff
@@ -338,21 +339,47 @@ func enableOrDisableServices(services imagecustomizerapi.Services, imageChroot *
 func loadOrDisableModules(modules imagecustomizerapi.Modules, imageChroot *safechroot.Chroot) error {
 	var err error
 
+	loadModuleMap := make(map[string]bool)
 	for _, module := range modules.Load {
-		logger.Log.Infof("Loading kernel module (%s)", module.Name)
-		moduleFileName := module.Name + ".conf"
-		moduleFilePath := filepath.Join(imageChroot.RootDir(), "/etc/modules-load.d/", moduleFileName)
-		err = file.Write(module.Name, moduleFilePath)
-		if err != nil {
+		if err := writeModuleLoadConfiguration(module, imageChroot); err != nil {
 			return fmt.Errorf("failed to write module load configuration: %w", err)
 		}
+		loadModuleMap[module] = true
+	}
+
+	var modulesWithOptions []string
+	for module, moduleOptions := range modules.Options {
+		if !loadModuleMap[module] {
+			// load the module in options list if it is not specified in load list
+			if err := writeModuleLoadConfiguration(module, imageChroot); err != nil {
```

Comment:
Why do we want to force the module to load if options are specified?

Reply:
if understand previous comment correctly "If you want to set kernel module options, you have to set it to be auto-enabled", so I thought in a yaml file, if a module is in options list but not in load list, we will also need to add a .conf in /etc/modules-load.d to load it, in combination with an options config in /etc/modprobe.d/module-options.conf

Reply:
It's possible that a user might want to set module options but continue to let the module load only if the correct hardware is present, instead of forcing the module to load on boot.

Reply:
make sense, then I will not put it in the list for boot time loading.